### ISSUE TITLE
Fails to initialize on darwin_arm64 architecture

### ISSUE THREAD

**Issue by mikluko** [OJFord/terraform-provider-wireguard#9]
Initialization of a workspace with Wireguard provider on Macbook with M1 chip results in error:

```
Error: Incompatible provider version
│
│ Provider registry.terraform.io/ojford/wireguard v0.1.3 does not have a package available for your
│ current platform, darwin_arm64.
│
│ Provider releases are separate from Terraform CLI releases, so not all providers are available for all
│ platforms. Other versions of this provider may have different platforms supported. 
```

Expected result: workspace initializes successfully.

**Comment by mikluko** [comment]
Released version was built with go 1.14 which never supported arm64. Building with go >= 1.16 should resolve this issue.

**Comment by OJFord** [comment]
Thanks for the report, I wasn't aware.

I've actually already upgraded to Go 1.17 I think it is (on mobile) but have been having some trouble with goreleaser. Once that's resolved, v0.2.x should work.

In the meantime you could clone, `go build`, and point terraform at your clone dir with `dev_overrides` in your tf config file.

I'll have another look at releasing later today.

**Comment by OJFord** [comment]
@mikluko I've finally resolved the problem signing releases, and released v0.2.0 & v0.2.1, both of which have darwin_arm64 builds, as will all releases going forwards.

Thanks again for bringing this to my attention.
